ext/openssl/extconf.rb

```diff
@@ -185,7 +185,7 @@ def find_openssl_library
 Logging::message "=== Checking done. ===\n"
 
 if openssl_3
-  if $warnflags.sub!(/-W\K(?=deprecated-declarations)/, 'no-')
+  if $warnflags && $warnflags.sub!(/-W\K(?=deprecated-declarations)/, 'no-')
```

Comment:
Since this part exists only in the file bundled with ruby, we don't have to backward compatibility.
What about the safe call?

```suggestion
  if $warnflags&.sub!(/-W\K(?=deprecated-declarations)/, 'no-')
```